app/src/components/trace/TraceTreeContext.tsx

```diff
@@ -0,0 +1,40 @@
+import React, {
+  createContext,
+  PropsWithChildren,
+  startTransition,
+  useCallback,
+  useContext,
+  useState,
+} from "react";
+
+export type TraceTreeContextType = {
+  isCollapsed: boolean;
+  setIsCollapsed: (theme: boolean) => void;
+};
+
+export const TraceTreeConnext = createContext<TraceTreeContextType | null>(
+  null
+);
+
+export function useTraceTree() {
+  const context = useContext(TraceTreeConnext);
+  if (context === null) {
+    throw new Error("useTheme must be used within a TraceTreeProvider");
+  }
+  return context;
+}
+
+export function TraceTreeProvider(props: PropsWithChildren) {
```

Comment:
why not use zustand here?

Reply:
because it's just one flag and there's no difference